streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableFilter.java

```diff
@@ -108,24 +92,45 @@ public void process(K key, Change<V> change) {
 
     }
 
-    private class KTableFilterValueGetter implements KTableValueGetter<K, V> {
+    @Override
+    public KTableValueGetterSupplier<K, V> view() {
+        // if the KTable is materialized, use the materialized store to return getter value;
+        // otherwise rely on the parent getter and apply filter on-the-fly
+        if (queryableName != null) {
+            return new KTableMaterializedValueGetterSupplier<>(queryableName);
+        } else {
+            return new KTableValueGetterSupplier<K, V>() {
```

Comment:
I would really like to share code with `KTableMapValues` . Can't we create an interface for the `computeValue()` method that both `KTableFilterValueGetter` and `KTableMapValueValueGetter` need to call? It's the exact some pattern for MapValue and Filter.

Reply:
I tried to do that, but it ended up not saving much LOC since we need to have a new interface class with both of them implementing it. So I ended up leaving this as is.

Reply:
Ok. Fine to leave as-is.